lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/client/REEFImplementation.java

```diff
@@ -89,8 +90,21 @@
 
   @Override
   public void close() {
+
+    LOG.log(Level.FINE, "Close REEF: shutdown jobs");
     this.runningJobs.closeAllJobs();
+
+    LOG.log(Level.FINE, "Close REEF: shutdown client");
     this.clientWireUp.close();
+
+    LOG.log(Level.FINE, "Close REEF: shutdown job submitter");
+    try {
+      this.jobSubmissionHandler.close();
```

Comment:
Do you see any error or improper shut down if we don't call jobSubmissionHandler.close()?